src/contracts/VaultHelper.sol

```diff
@@ -1,359 +1,444 @@
+        if (!VAULT_FACTORY.isEntity(vault)) revert VaultHelper__InvalidVault(vault);
+        if (collateral == address(0)) revert VaultHelper__ZeroAddress("collateral");
+        if (underlying == address(0)) revert VaultHelper__ZeroAddress("underlying");
+        if (user == address(0)) revert VaultHelper__InvalidUser(user);
+        if (amount == 0) revert VaultHelper__InvalidAmount(amount);
+
+        // --- Step 1: Measure balance before the transfer ---
+        uint256 balanceBefore = IERC20(underlying).balanceOf(address(this));
+
+        // --- Step 2: Pull underlying tokens from msg.sender into proxy ---
+        IERC20(underlying).safeTransferFrom(msg.sender, address(this), amount);
+
+        // --- Step 3: Calculate the actual amount received ---
+        uint256 actualAmount = IERC20(underlying).balanceOf(address(this)) - balanceBefore;
+
+        // --- Step 4: Approve collateral contract to pull the actual amount from proxy ---
+        IERC20(underlying).forceApprove(collateral, actualAmount);
+
+        // --- Step 5: Deposit actual amount into collateral, minting collateral tokens to proxy ---
+        collateralAmount = IDefaultCollateral(collateral).deposit(address(this), actualAmount);
+        if (collateralAmount == 0) revert VaultHelper__ZeroCollateralMint(actualAmount, collateral);
+        IERC20(underlying).forceApprove(collateral, 0);
+
+        // --- Step 6: Approve vault to pull collateral tokens from proxy ---
+        IERC20(collateral).forceApprove(vault, collateralAmount);
+
+        // --- Step 7: Deposit collateral into vault, minting the vault shares tokens to proxy ---
+        (, uint256 vaultSharesMinted) = IVaultTokenized(vault).deposit(address(this), collateralAmount);
+        if (vaultSharesMinted == 0) revert VaultHelper__ZeroVaultSharesMinted(collateralAmount);
+        IERC20(collateral).forceApprove(vault, 0);
+
+        // --- Step 8: Approve LSTWrapper to pull vault shares from proxy ---
+        IERC20(vault).forceApprove(lstWrapper, vaultSharesMinted);
+
+        // --- Step 9: Deposit vault shares into LSTWrapper on behalf of the receiver (`user`) ---
+        lstSharesMinted = ILSTWrapper(lstWrapper).deposit(vaultSharesMinted, user);
+        if (lstSharesMinted == 0) revert VaultHelper__ZeroLSTWrapperSharesMinted(lstSharesMinted);
+        IERC20(lstWrapper).forceApprove(lstWrapper, 0);
+
+        return (collateralAmount, lstSharesMinted);
+    }
+
+    function withdrawFromWrappedVault(
+        address vault,
+        address user,
+        address lstWrapper,
+        uint256 amount
+    ) external nonReentrant returns (uint256 collateralAmount, uint256 lstSharesBurned) {
+        if (vault == address(0)) revert VaultHelper__ZeroAddress("vault");
+        if (!VAULT_FACTORY.isEntity(vault)) revert VaultHelper__InvalidVault(vault);
 
-            pendingWithdraws = new PendingWithdraw[](count);
+        lstSharesBurned = ILSTWrapper(lstWrapper).withdraw(amount, address(this), user);
```

Comment:
You are mixing shares and assets here. 

For `LSTWrapper`, `asset()` is the vault share token (IVaultTokenized). 
The name`lstSharesBurned` is representing amount of vault shares (as its withdrawn), not lst shares as it suggests.